crates/ruff_linter/src/rules/pyupgrade/rules/outdated_version_block.rs

```diff
@@ -88,7 +88,9 @@ pub(crate) fn outdated_version_block(checker: &mut Checker, stmt_if: &StmtIf) {
         match comparison {
             Expr::Tuple(ast::ExprTuple { elts, .. }) => match op {
                 CmpOp::Lt | CmpOp::LtE => {
-                    let version = extract_version(elts);
+                    let Some(version) = extract_version(elts) else {
```

Comment:
What's preventing us from matching on `Int` directly and create a diagnostic if the value is too large (because the `CmpOp::GtE` likely fails? 

Which brings up a good point. We should implement `PartialOrd` and `Ord` for all types for which we can safely support it.

Reply:
I don't think you can implement `Ord` for mixed types, only `PartialOrd`?

Reply:
Also, I don't think `CmpOp::GtE` would be likely to fail, since Python does support numbers larger than this.